examples/data_preprocessing/mimic_dataset/prepare_mimic_cxr.ipynb

```diff
@@ -0,0 +1,1862 @@
+    "        \n",
+    "    def __len__(self):\n",
+    "        'Denotes the total number of samples'\n",
+    "        return len(self.path)\n",
+    "    \n",
+    "    def __getitem__(self, index):\n",
+    "        'Generates one sample of data'\n",
+    "        # Select sample\n",
+    "        image = Image.open(self.path[index,3].replace(\".dcm\", \".jpg\")).convert('RGB')\n",
+    "        X = self.transform(image)\n",
+    "        if self.load_labels: # for the second approach with finetuning\n",
+    "            label = self.path[index,5]      \n",
+    "            return X, torch.tensor(label)\n",
+    "        else:\n",
+    "            return X # for the first approach without labels\n",
+    "    \n",
+    "    transform = transforms.Compose([\n",
+    "        transforms.Resize(256),\n",
+    "        transforms.CenterCrop(224),\n",
+    "        transforms.ToTensor(),\n",
+    "        transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])])\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We load the ResNet50 model and remove the last layer, so the output is then the second last layer with dimension 1x2048. These are our features. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 48,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model = models.resnet50(pretrained=True)\n",
+    "modules = list(model.children())[:-1]\n",
+    "model=torch.nn.Sequential(*modules)\n",
+    "for p in model.parameters():\n",
+    "    p.requires_grad = False\n",
+    "    \n",
+    "model.eval()\n",
+    "# apply modified resnet50 to data\n",
+    "dataloaders = DataLoader(mimicDataset(input_list[:n,:]), batch_size=n,num_workers=0)\n",
+    "    \n",
+    "data = next(iter(dataloaders))\n",
+    "with torch.no_grad():\n",
+    "    features_var = model(data)\n",
+    "    features = features_var.data \n",
+    "    train_X = features.reshape(n,2048).numpy()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally, we save the image feature matrix."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 49,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "['train_X.lib']"
+      ]
+     },
+     "execution_count": 49,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "dump(train_X, X)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Second approach: Finetuning a pretrained CNN (with weak CheXpert labels)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In the second approach of image encoding, we use the concept of transfer learning. \n",
+    "Therefore, we take a pretrained CNN and continue training the model with our data. This saves a lot of time and often leads to satisfying results. \n",
```

Comment:
```suggestion
    "Therefore, we take a CNN pretrained on ImageNet and continue training the model with our data. This saves a lot of time and often leads to satisfying results. \n",
```

Reply:
Also I don't understand this sentence: "This saves a lot of time and often leads to satisfying results.". Since the finetuning takes some time, this approach doesn't save time compared to the first approach. Moreover, it's important to mention that images from ImageNet are very different are different from medical images, so this finetuning can help improve the results.